external/urlfrontier/src/test/java/com/digitalpebble/stormcrawler/urlfrontier/StatusBoldTest.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.digitalpebble.stormcrawler.urlfrontier;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.digitalpebble.stormcrawler.Metadata;
+import com.digitalpebble.stormcrawler.TestOutputCollector;
+import com.digitalpebble.stormcrawler.TestUtil;
+import com.digitalpebble.stormcrawler.persistence.Status;
+import io.grpc.ManagedChannel;
+import java.util.HashMap;
+import java.util.concurrent.*;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.tuple.Tuple;
+import org.junit.*;
+import org.junit.rules.Timeout;
+
+public class StatusBoldTest {
+    private StatusUpdaterBolt bolt;
+    private TestOutputCollector output;
+    private URLFrontierContainer urlFrontierContainer;
+
+    private ManagedChannel managedChannel;
+
+    private static final String persistedKey = "somePersistedKey";
+
+    private static ExecutorService executorService;
+
+    @Rule public Timeout globalTimeout = Timeout.seconds(60);
+
+    @BeforeClass
+    public static void beforeClass() {
+        executorService = Executors.newFixedThreadPool(2);
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        executorService.shutdown();
+        executorService = null;
+    }
+
+    @Before
+    public void before() {
+
+        urlFrontierContainer = new URLFrontierContainer("crawlercommons/url-frontier:2.1");
```

Comment:
I'm not that accustomed to poms, can i specify the testcontainer-version in the root-pom and use is in a module pom?

Reply:
Thats why i applied it in the "root-pom" https://github.com/DigitalPebble/storm-crawler/pull/980#discussion_r912806211

Reply:
Ah, forget that comment. This was about getting the version of URLFrontier
(I'll leave that link here, so that I can find it when I rework it. https://github.com/DigitalPebble/storm-crawler/blob/master/external/elasticsearch/src/test/java/com/digitalpebble/stormcrawler/elasticsearch/bolt/StatusBoltTest.java#L65)

Reply:
> I'm not that accustomed to poms, can i specify the testcontainer-version in the root-pom and use is in a module pom?

yes, this is done through _<dependency-management>_, see https://github.com/DigitalPebble/storm-crawler/blob/master/pom.xml#L262